axiom/nr_php_packages.c

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2022 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <math.h>
+#include <stdlib.h>
+#include <stdio.h>
+
+#include "nr_axiom.h"
+#include "nr_limits.h"
+#include "nr_php_packages.h"
+#include "util_memory.h"
+#include "util_minmax_heap.h"
+#include "util_vector.h"
+#include "util_hashmap.h"
+#include "util_hashmap_private.h"
+#include "util_strings.h"
+
+nr_php_package_t* nr_php_package_create(char* name, char* version) {
+  nr_php_package_t* p = NULL;
+
+  if (name == NULL || version == NULL) {
+    return NULL;
+  }
+
+  p = (nr_php_package_t*)nr_malloc(sizeof(nr_php_package_t));
+
+  p->package_name = nr_strdup(name);
+  p->package_version = nr_strdup(version);
+
+  return p;
+}
+
+void nr_php_package_destroy(nr_php_package_t* p) {
+  if (p != NULL) {
+    nr_free(p->package_name);
+    nr_free(p->package_version);
+    nr_free(p);
+  }
+}
```

Comment:
To avoid a situation of accessing an invalid memory by referencing a pointer to memory that has been freed, I suggest `nr_php_package_destroy` to set `p` to `NULL` (like [here](https://github.com/newrelic/newrelic-php-agent/blob/4713dc219062b00e6771bbd5dfba92f5f7a0a60c/axiom/nr_log_event.c#L21-L38)):
```suggestion
void nr_php_package_destroy(nr_php_package_t** ptr) {
  nr_php_package_t* p;

  if ((NULL == ptr) || (NULL == *ptr)) {
    return;
  }

  p = *ptr;
  nr_free(p->package_name);
  nr_free(p->package_version);

  nr_realfree((void**)ptr);
}
```

Reply:
After discussion, we will not make this change because it directly impacts the `nr_php_packages_add_package()` function since we are passing the `nr_php_package_destroy()` function as a destructor when `nr_hashmap_create()` is called and this change would cause the signatures to not match. It is also important to note that `nr_php_package_destroy` is only used as a dtor_func callback in the hashmap and not used anywhere else.